### ISSUE TITLE
[Feature Request] Missing ActionScaling and FlattenAction 

### ISSUE THREAD

**Issue by duburcqa** [pytorch/rl#1209]
## Motivation

In general, the action space may be a dict instead of a plain vector, and is not normalized. Even for a classic `gym` environment, action wrappers performing action flattening and normalization according the min/max spec are not provided by the standard API, but are implemented by many libraries.

## Solution

First, add a normalization transform `ActionScaling` to `torchrl` that relies on the spec to compute `loc` and `scale` unlike `ObservationNorm` nor moving average like `VecNorm`. I would advice raising an exception for completely or partially unbounded action spaces. Second, add an action flattening transform `FlattenAction` foing very much the same as `FlattenObservation`.

## Alternatives

Keep relying on flattening and normalization wrappers at env-level provided by any other library.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

**Comment by vmoens** [comment]
Got it sounds about right!
We could also have a method in observation norm to compute loc and scale based on the spec now that you mention that.

**Comment by duburcqa** [comment]
Yes, but it is much less common for observation space to be fully bounded. In any case, the observation distribution under the policy rarely span the whole space in practice, so it would probably be much less frequently used.

**Comment by PSXBRosa** [comment]
Hello,

I would like to work on this Issue, is anyone already working on it?

Just for making sure I interpreted the issue correctly, the goal is implementing those two wrappers for the action space, is that right?

**Comment by duburcqa** [comment]
Yes it is ! I don't think someone is working on it for now.

**Comment by pxe3** [comment]
Hello, @vmoens. I apologize for the necro ping, is this issue still available?

I see that it has been some time since it was last worked on, but it was never resolved. If so, I would be happy to open a PR completing it. This would be my first open-source contribution (of many).

Let me know!

**Comment by Drakit0** [comment]
I've split this into two small, reviewable PRs:

- #3765 - Add `ActionScaling` transform (part 1)
- #3766 - Add `FlattenAction` transform (part 2)

Both follow the existing transform conventions:

- `ActionScaling`: spec-driven affine normalization. Exposes a `[-1, 1]` (or `[0, 1]`) action space to the policy and rescales actions to the env's bounds on the inv direction. Raises `RuntimeError` for fully or partially unbounded specs (covering both `Unbounded` and bounded specs whose entries match the `torch.finfo` extremes used internally by `Unbounded`).
- `FlattenAction`: flattens adjacent action dims, mirroring `FlattenObservation`. The inv direction unflattens back to the env-side shape. Both flat and nested action keys are supported.

Each PR ships with unit tests (`TransformBase` subclass plus class-specific cases) and was validated locally via `pytest test/transforms/test_action_transforms.py -x` on Python 3.12 / Torch 2.8.

**Comment by vmoens** [comment]
Implemented by #3765 (`ActionScaling`) and #3766 (`FlattenAction`), both merged. Closing this feature request as completed. Thanks to everyone who helped carry it over the line.
